nexus_constructor/model/dataset.py

```diff
@@ -4,20 +4,52 @@
 
 from nexus_constructor.common_attrs import CommonAttrs
 from nexus_constructor.model.node import Node
-from nexus_constructor.model.value_type import ValueType
+from nexus_constructor.model.value_type import ValueType, VALUE_TYPE
+
+
+def _get_human_readable_type(new_value: Any):
```

Comment:
Use composition here and contain attributes within `Node` and all of its helpers rather than using inheritance. Then we can bin this code off and require the dtype to be completely required rather than guessing it from the value.